src/main/daml/Daml/Finance/Instrument/Option/Election.daml

```diff
@@ -0,0 +1,100 @@
+-- Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Instrument.Option.Election where
+
+import DA.Set qualified as S (singleton)
+import Daml.Finance.Interface.Instrument.Option.Election qualified as Election (Apply(..), ApplyElection(..), HasImplementation, I, View(..))
+import Daml.Finance.Interface.Instrument.Option.Election.Factory qualified as Factory (Create(..), F, HasImplementation, View(..))
+import Daml.Finance.Interface.Lifecycle.Event qualified as Event (I, View(..))
+import Daml.Finance.Interface.Types.Common.Types (Id(..), InstrumentKey, PartiesMap)
+import Daml.Finance.Interface.Util.Disclosure qualified as Disclosure (I, View(..), flattenObservers)
+import Daml.Finance.Util.Disclosure (addObserversImpl, removeObserversImpl, setObserversImpl)
+
+-- | Type synonym for `Election`.
+type T = Election
+
+instance Election.HasImplementation T
+-- | An election, such as the exercise of an option.
+template Election
```

Comment:
Is this any different than the election implementation provided with the Generic instrument?  Same for the `Election` interface

Reply:
Probably not, I think I only had to change the Lifecycle Rule in the end. 

I was not sure where this code should sit, so I opened this: https://github.com/digital-asset/daml-finance/issues/661

Reply:
Understood, thanks. Would it make sense to extend the existing rule for dynamic claim instruments and add the implementation of `Exercisable` in there?

Reply:
Possibly, just not sure when. At the moment, I have a simplified implementation that covers the functionality currently needed, but the Rule will probably have to be extended soon due to new option types. Maybe keeping them separate for now (easier to extend without impacting / bumping other packages) and merge later on?

Reply:
I added an issue to track this: https://github.com/digital-asset/daml-finance/issues/666